framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileElementTypes.java

```diff
@@ -313,6 +321,81 @@ public AnnotatedTypeMirror getAnnotatedTypeMirror(Element e) {
         return Collections.emptySet();
     }
 
+    /**
+     * Returns the method type of the most specific fake override for the given element, when used
+     * as a member of the given type.
+     *
+     * @param elt element for which annotations are returned
+     * @param receiverType the type of the class that contains member (or a subtype of it)
+     * @return the most specific AnnotatedTypeMirror for {@code elt} that is a fake override, or
+     *     null if none
+     */
+    public AnnotatedTypeMirror getFakeOverride(Element elt, AnnotatedTypeMirror receiverType) {
+        if (parsing) {
+            throw new BugInCF("parsing while calling getFakeOverride");
+        }
+
+        if (elt.getKind() != ElementKind.METHOD) {
+            return null;
+        }
+
+        ExecutableElement method = (ExecutableElement) elt;
+
+        TypeMirror methodReceiverType = method.getReceiverType();
+        if (methodReceiverType != null && methodReceiverType.getKind() == TypeKind.NONE) {
+            return null;
+        }
+
+        // This is a list of pairs of (where defined, method type) for fake overrides.  The second
+        // element of each pair is currently always an AnnotatedExecutableType.
+        List<Pair<TypeMirror, AnnotatedTypeMirror>> candidates =
+                annotationFileAnnos.fakeOverrides.get(elt);
+
+        if (candidates == null || candidates.isEmpty()) {
+            return null;
+        }
+
+        TypeMirror receiverTypeMirror = receiverType.getUnderlyingType();
+
+        // A list of fake receiver types.
+        List<TypeMirror> applicable = new ArrayList<>();
+        for (Pair<TypeMirror, AnnotatedTypeMirror> candidatePair : candidates) {
+            TypeMirror fakeLocation = candidatePair.first;
+            AnnotatedExecutableType candidate = (AnnotatedExecutableType) candidatePair.second;
+            if (factory.types.isSameType(receiverTypeMirror, fakeLocation)) {
+                return candidate;
+            } else if (factory.types.isSubtype(receiverTypeMirror, fakeLocation)) {
+                applicable.add(fakeLocation);
+            }
+        }
+
+        if (applicable.isEmpty()) {
+            return null;
+        }
+        TypeMirror fakeReceiverType =
+                TypesUtils.mostSpecific(applicable, factory.getProcessingEnv());
+        if (fakeReceiverType == null) {
```

Comment:
I don't understand how there can be no "most specific" fake override.  Can you explain how this case occurs?

Reply:
I thought it was possible to write
```java
class A { public int m() { return 0; } }
interface I { int m(); }
class B extends A implements I { }
```
but that is not legal Java.  I will change the `UserError` to `BugInCF`.

Reply:
`javac` compiles that code.